src/internal/document_api/mod.rs

```diff
@@ -1054,4 +1362,31 @@ mod tests {
         let deduplicated_grants = dedupe_grants(&grants_w_dupes);
         assert_that!(&deduplicated_grants.len(), eq(2))
     }
+
+    #[test]
+    fn roundtrip_encrypted_dek_proto() {
```

Comment:
This test doesn't assert anything about the code you wrote above. Round trip would work even if you put everything in the wrong place. 

We need a test that asserts the translation is correct.

Reply:
test was added